README.markdown

```diff
@@ -186,8 +186,33 @@ instance FiniteBits Foo where
 #endif
 ```
 
-This is a very unsatisfactory solution, and for this reason, we do not pursue
-it. For similar reasons, we do not backport data types.
+Another approach is to backport the API from `base-4.7.0.0` and to declare
+additional methods outside of the class:
+
+```
+#if MIN_VERSION_base(4,7,0) && !(MIN_VERSION_base(4,8,0))
+countLeadingZeros :: FiniteBits b => b -> Int
+countLeadingZeros = {- default implementation #-}
+#endif
+```
+
+The situation is only sligtly better for classes which exist across all `base` version,
+but have grown their API. For example, it's tempting to define
+
+```haskell
+#if !(MIN_VERSION_base(4,8,0))
+displayException :: Exception e => e -> String
+displayException = show
+#endif
+```
+
+As with previous approach, you won't be able to define new members of the type
+class without CPP guards. In other words, non-CPP approach would limit
```

Comment:
In other words, _the_ non-CPP approach would limit uses to _the_ lowest common denominator.